nightfall-deployer/README.md

```diff
@@ -0,0 +1,180 @@
+# Deployment Procedure
+The deployment procedure is the process of compiling the Smart Contracts, deploying them to the desired Network and performing the circuits' setup.
+
+To perfom a deployment follow the instructions bellow:
+1. Create a .env file from the template file `.env.deployment.template` and change the name to match your network. e.g. `.env.deployment.my_network`. This 
+  file has settings that will be used for performing the deployment. One should change them accordingly with the desired values. The following variables are 
+  important to be set:
+  - `ETH_NETWORK`
+  - `ETH_ADDRESS`
+  - `ETH_PRIVATE_KEY`
+  - `BLOCKCHAIN_URL`
+  - `MULTISIG_SIGNATURE_THRESHOLD`
+  - `MULTISIG_APPROVERS`
+
+2. One entry should be added under `networks` in the `truffle-config.js` file, similar to the following:
+  ```
+      [NETWORK_NAME]: {
+        provider: () => new HDWalletProvider(config.ETH_PRIVATE_KEY, config.BLOCKCHAIN_URL),
+        network_id: [NETWORK_ID],
+        networkCheckTimeout: 1000000000,
+        timeoutBlocks: 2000,
+        confirmations: 2,
+        skipDryRun: true,
+        websockets: true,
+        gasPrice: config.WEB3_OPTIONS.gasPrice,
+        gas: config.WEB3_OPTIONS.gas,
+      },
+  ```
+  Replace the identified placeholder with proper values:
+    - `NETWORK_NAME`: with your network id. This can be any valid string, without spaces, since it doesn't conflict with the existing ones; 
+    - `NETWORK_ID`: An integer number representing the ID of your network;
+    - The other values are not that relevant, but the network in question might require some specifics values for them, then one should 
+    verify if these values need to be changed.
+
+  The final result would be similar to the following example:
+  ```
+      mumbai: {
+        provider: () => new HDWalletProvider(config.ETH_PRIVATE_KEY, config.BLOCKCHAIN_URL),
+        network_id: 80001,
+        networkCheckTimeout: 1000000000,
+        timeoutBlocks: 2000,
+        confirmations: 2,
+        skipDryRun: true,
+        websockets: true,
+        gasPrice: config.WEB3_OPTIONS.gasPrice,
+        gas: config.WEB3_OPTIONS.gas,
+      },
+  ```
+3. In the `config/defaults.js` file it is required to set some configurations related to the deployment one desires to perform:
+  - Find the `ENVIRONMENTS` object and add a new entry for the network ones disire to deploy to. Like in the `Step 2.`, the placeholders should be
+    replaced by proper values. To be consistent, the same values used in the previous step can be used again here. The `WEB3_WS_URL` is the Websocket
+    endpoint used to communicate with the blockchain. e.g.
```

Comment:
SUGGESTION

3. Like in `Step 2.`, add a new entry for the network one desires to deploy to under `ENVIRONMENTS` in the `config/defaults.js` file. Placeholders should be replaced by proper values. To be consistent, the same values used in the previous step can be used again here. The `WEB3_WS_URL` is the Websocket endpoint used to communicate with the blockchain. e.g.